function_app.py

```diff
@@ -33,6 +33,15 @@
     connection=EVENTHUB_CONNECTION_VAR,  # the parameter expects the env var name not the value.
     cardinality="many",
 )
+# Configures an exponential backoff retry strategy in the trigger from eventhub to the function.
+# When eventuhb executes the function, it will not commit a checkpoint until all retries are exhausted,
+# and then progress in that partition is restarted.
+@app.retry(
+    strategy="exponential_backoff",
+    max_retry_count="10",
+    minimum_interval="00:00:01", # format followed is HH:mm:ss
+    maximum_interval="00:00:07",
+)

```

Comment:
Sorry, I've missed your PR. I just added my retry policy myself: https://github.com/grafana/azure_eventhub_to_loki/blob/40faeefbbcb30da917ae97bf2507155a751a4279/function_app.py#L40-L45

I suggest the following
```suggestion
# Configures an exponential backoff retry strategy in the trigger from eventhub to the function.
# When eventuhb executes the function, it will not commit a checkpoint until all retries are exhausted,
# and then progress in that partition is restarted.
@app.retry(
    strategy="exponential_backoff",
    max_retry_count=MAX_RETRY_COUNT,
    minimum_interval=MINIMUM_INTERVAL,
    maximum_interval=MAXIMUM_INTERVAL,
)
```

And earlier
```python
MAX_RETRY_COUNT: Final[str] = "3"
MINIMUM_INTERVAL: Final[str] = "00:00:01"
MAXIMUM_INTERVAL: Final[str] = "00:00:10"
```

And then in the except block
```python
        if context.retry_context.retry_count == context.retry_context.max_retry_count:
            logging.exception(
                "failed to process event %d times. Giving up.",
                context.retry_context.retry_count + 1,
            )
        else:
            logging.exception(
                "failed to process event %d times. Retrying...",
                context.retry_context.retry_count + 1,
            )
            raise
```